apps/shipments/permissions.py

```diff
@@ -84,6 +79,35 @@ def has_object_permission(self, request, view, obj):
            self.has_moderator_permission(request, obj)
 
 
+class IsDeviceOwnerOrShared(IsShipmentOwnerMixin,
```

Comment:
I think we can DRY this up, not sure we need a new permission class to handle this. We need to look up shipment by device_id and then check the user's permissions for that shipment, right?

So I think we could add a check to `IsOwnerOrShared`, similar to the nested_shipment check, except it would be nested_device and lookup the shipment by device_id, and return self.has_object_permission.

Adds a few lines to IsOwnerOrShared but overall I think it's cleaner